lib/datadog/core/configuration/settings.rb

```diff
@@ -541,8 +541,9 @@ def initialize(*_)
 
           # Tune remote configuration polling interval.
           #
-          # @default `DD_REMOTE_CONFIGURATION_POLL_INTERVAL_SECONDS` environment variable, otherwise `5.0` seconds.
+          # @default `DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS` environment variable, otherwise `5.0` seconds.
           # @return [Float]
+          # @!visibility private
```

Comment:
Should we document here a stronger "don't tune this, this setting is only used for internal validation and changing for other use-cases it is unsupported" or something like that?

Reply:
Since this is only used from system tests and system tests only use ENV variables, we might get away by not exposing the setting at all. 

We could read the ENV variable. here: https://github.com/DataDog/dd-trace-rb/blob/86f9593268706f2b1a288a6ebf9e6dad29ff2e0d/lib/datadog/core/remote/component.rb#L33 and if it not present default to `5.0`

That way, there is no need to document that is private we just make it private 😎

Reply:
That's a good point, although you lose the consistency of always going through the settings. I guess we could at least move such settings to an `internal` or even `debug` group, if we wanted to still keep them.

Reply:
I live having it go through `settings` for consistency. It will get better with configuration validation 😉.

I've added stronger wording for it being private.